Add Web Platform Support for 3.x

kuasha420 opened the pull request:
<!-- Thanks for submitting a pull request! We appreciate you spending the time to work on these changes. Please follow the template so that the reviewers can easily understand what the code changes affect -->

# Summary

<!--
Explain the **motivation** for making this change: here are some points to help you:

* What issues does the pull request solve? Please tag them so that they will get automatically closed once the PR is merged
* What is the feature? (if applicable)
* How did you implement the solution?
* What areas of the library does it impact?
-->

I have finally got some time and motivation to port over my previous web implementation for the latest tag. Please have a look and see if this is something you'd like to merge to the main tree. Don't hesitate to let me know if anything needs changing. 

## Test Plan

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->

I have created an example repo showcasing this. 

[link](https://github.com/kuasha420/rn-bootsplash-web-example)

`yarn && yarn web`

### What's required for testing (prerequisites)?

```sh
# Web Example
yarn react-native generate-bootsplash assets/bootsplash_logo_original.png \
  --background-color=F5FCFF \
  --logo-width=100 \
  --assets-path=assets \
  --webroot-path=web
```

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅   |
| Android |    ✅    |
| Web |    ✅   |

## Checklist

<!-- Check completed item, when applicable, via: [X] -->

- [ ] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

kuasha420 commented:
@zoontek will you kindly have a look at this? Also, I see no CHANGELOG.md in the project to mention it and not sure about bloating the example with the web integration webpack stuffs. 

Cheers.

zoontek commented:
@kuasha420 Thank you! I will review it next week. It shouldn't be an issue for the v4 as the new version update mainly the android part.

zoontek commented on src/index.ts:
```diff
@@ -1,13 +1,15 @@
-import { NativeModules } from "react-native";
+import { NativeModules } from "./react-native";
```
I think it's a mistake.

kuasha420 commented:
@zoontek I have fixed the conflicts and refactored the markup to significantly simplify the html/css code. Also added support for "homepage" field that would  help with path problems. Please have a when/if you have the time and let me know if anything needs changing still.

Cheers. 

PS: updated the example repo as well.

itsaladin commented:
LGTM

zoontek commented:
@dmitrymatensemble This is currently not a priority. I would rather have proper iOS 15, Android 12 support + support the new react-native architecture (JSI) before having web support.

zoontek commented:
The PR is quite clean, thanks! ✨
Work will continue here: https://github.com/zoontek/react-native-bootsplash/pull/278
